triggers/service/auth/src/main/scala/com/daml/oauth/middleware/Server.scala

```diff
@@ -88,8 +104,10 @@ object Server extends StrictLogging {
             responseType = "code",
             clientId = config.clientId,
             redirectUri = toRedirectUri(request.uri),
-            scope = Some(login.claims),
-            state = Some(requestId.toString))
+            scope = Some(login.claims.toQueryString),
+            state = Some(requestId.toString),
+            audience = Some("https://daml.com/ledger-api")
```

Comment:
I think I’ve mentioned it before but I expect we need to make the mapping here pretty configurable. For auth0 you are very flexible in defining how scopes are mapped to the token but this isn’t necessarily true for all auth providers and some of them also use arbitrary other arguments. One idea would be to have a jsonnet template (or whatever other template format you come up with, I’m not familiar with what people use for this) to basically define the whole request body here as users need to.

Definitely not something for this PR, but something to keep in mind.

Reply:
Yes, this is auth0 specific. What I've seen in other places is presets for known auth providers. Templating is a nice idea, I'll keep it in mind.